client/jetpack-onboarding/steps/stats.jsx

```diff
@@ -86,14 +86,17 @@ class JetpackOnboardingStatsStep extends React.Component {
 	}
 
 	renderActionTile() {
-		const { isConnected, siteUrl, translate } = this.props;
+		const { basePath, isConnected, siteSlug, siteUrl, stepName, translate } = this.props;
 		const headerText = translate( 'Keep track of your visitors with Jetpack.' );
 		const subHeaderText = translate(
 			'Keep an eye on your success with simple, concise, and mobile-friendly stats. ' +
 				'Get updates on site traffic, successful posts, site searches, and comments, all in real time.'
 		);
+
 		const connectUrl = addQueryArgs(
 			{
+				from: 'jpo',
+				redirectAfterAuth: [ basePath, stepName, siteSlug ].join( '/' ),
 				url: siteUrl,
 				// TODO: add a parameter to the JPC to redirect back to this step after completion
 				// and in the redirect URL include the ?action=activate_stats parameter
```

Comment:
For direct auth we might want to hit the Jetpack wp-admin directly, and not `/jetpack/connect` - similar to how it's done in #21755. We could do that in a subsequent PR though if you prefer, but I'd say those are pretty related (because what we're doing here will eventually end in the same place).